### PR TITLE
Fix false positive 'unreachables'

### LINKED ISSUE #2259: False positive for unreachable_code_linter() on inline expressions
```r
lint("function(x) if (x > 2) stop() else x\n", unreachable_code_linter())
# <text>:1:31: warning: [unreachable_code_linter] Code and comments coming after a return() or stop() should be removed.
# function(x) if (x > 2) stop() else x
#                               ^~~~
```

This may be a regression; cc @MEO265.

### PR DESCRIPTION
**Pull request by MEO265**
Fixes #2259

### CODE REVIEW

**Comment by MEO265** [comment]
@MichaelChirico I know that I still have to create the NEWS entry, but the version has not been adjusted since the release. Should this happen through an extra PR or should it have been done automatically through the pipeline?

**Comment by MichaelChirico** [comment]
> @MichaelChirico I know that I still have to create the NEWS entry, but the version has not been adjusted since the release. Should this happen through an extra PR or should it have been done automatically through the pipeline?

Thanks, I just pinged Jim to push the CRAN commit here on GitHub.

File: tests/testthat/test-unreachable_code_linter.R [hunk 1]
```diff
@@ -597,6 +597,50 @@ test_that("function shorthand is handled", {
   )
 })
 
+test_that("Do not lint inline else after stop", {
+
+  expect_lint(
+    "if(x > 3L) stop() else x + 3",
```
Comment (AshesITR) [inline tests/testthat/test-unreachable_code_linter.R:603]: Irrelevant to the tests, but do you mind de-linting the tested code by adding a space between `if` and `(`?

Comment (MEO265) [reply tests/testthat/test-unreachable_code_linter.R]: I have done it

**Comment by MEO265** [comment]
> > @MichaelChirico I know that I still have to create the NEWS entry, but the version has not been adjusted since the release. Should this happen through an extra PR or should it have been done automatically through the pipeline?
> 
> Thanks, I just pinged Jim to push the CRAN commit here on GitHub.

Thanks, but shouldn't the version number be incremented or given the postfix ".9000" to avoid being confused with the one from CRAN?

**Comment by MichaelChirico** [comment]
Yep, that's #2264

**Review by MichaelChirico — APPROVED** [review]

**Comment by MichaelChirico** [comment]
LGTM, thanks!

**Pull request merged**